src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -188,6 +187,52 @@ private static void AddLinuxContainerServices(this IServiceCollection services)
                 var nullMetricsLogger = s.GetService<ILogger<NullMetricsPublisher>>();
                 return new NullMetricsPublisher(nullMetricsLogger);
             });
+
+            services.AddSingleton<IMeshInitServiceClient>(s =>
+            {
+                var environment = s.GetService<IEnvironment>();
+                if (environment.IsLinuxConsumption())
+                {
+                    var httpClient = s.GetService<HttpClient>();
+                    var logger = s.GetService<ILogger<MeshInitServiceClient>>();
+                    return new MeshInitServiceClient(httpClient, environment, logger);
+                }
+
+                var nullLogger = s.GetService<ILogger<NullMeshInitServiceClient>>();
+                return new NullMeshInitServiceClient(nullLogger);
+            });
+
+            services.AddSingleton<LinuxFunctionExecutionActivityPublisher>(s =>
+            {
+                var environment = s.GetService<IEnvironment>();
+                var logger = s.GetService<ILogger<LinuxFunctionExecutionActivityPublisher>>();
+                var meshInitServiceClient = s.GetService<IMeshInitServiceClient>();
+                var standbyOptions = s.GetService<IOptionsMonitor<StandbyOptions>>();
+                return new LinuxFunctionExecutionActivityPublisher(standbyOptions, meshInitServiceClient, environment, logger);
+            });
+
+            services.AddSingleton<IHostedService>(s =>
+            {
+                var environment = s.GetService<IEnvironment>();
+                if (environment.IsLinuxConsumption())
+                {
+                    return s.GetRequiredService<LinuxFunctionExecutionActivityPublisher>();
+                }
+
+                return NullHostedService.Instance;
+            });
+
+            services.AddSingleton<ILinuxFunctionExecutionActivityPublisher>(s =>
+            {
+                var environment = s.GetService<IEnvironment>();
```

Comment:
This doesn't seem right. You have two separate blocks for registering the same LinuxFunctionExecutionActivityPublisher, for IHostedService/ILinuxFunctionExecutionActivityPublisher